### ISSUE TITLE
Improve error message on invalid options

### ISSUE THREAD

**Issue by mfn** [FriendsOfPHP/PHP-CS-Fixer#5557]
## Feature request
For example, when I migrated to 3b1, I missed that for `no_extra_blank_lines` the `useTrait` in `tokens` isn't valid anymore.

However, the only error I got from the tool was:
```
In AbstractFixer.php line 158:

  [no_extra_blank_lines] Invalid configuration: The option "tokens" with value array is invalid.


In OptionsResolver.php line 990:

  The option "tokens" with value array is invalid.
```
I.e. it known something is invalid, but doesn't tell what :) What be great if that invalid value could be mentioned

```
$ ./php-cs-fixer-v2.phar fix --version  
PHP CS Fixer 3.0.0-beta.1  by Fabien Potencier and Dariusz Ruminski (22e1a37)

$ ./php-cs-fixer-v2.phar fix --rules='{"no_extra_blank_lines":{"tokens": ["invalid"]}}'
Loaded config default.

In AbstractFixer.php line 158:
                                                                                                  
  [no_extra_blank_lines] Invalid configuration: The option "tokens" with value array is invalid.  
                                                                                                  

In OptionsResolver.php line 990:
                                                    
  The option "tokens" with value array is invalid.  
                                                    

fix [--path-mode PATH-MODE] [--allow-risky ALLOW-RISKY] [--config CONFIG] [--dry-run] [--rules RULES] [--using-cache USING-CACHE] [--cache-file CACHE-FILE] [--diff] [--format FORMAT] [--stop-on-violation] [--show-progress SHOW-PROGRESS] [--] [<path>...]
```
(I never though the `--rules` supporting JSON would be useful, yet here I am 😄 )

**Comment by keradus** [comment]
would you like to provide a PR for this improvement, @mfn ?

**Comment by mfn** [comment]
I tried to understand the code but…

This is the part where the decision is made the any of the provided `$values` is no part of the `$this->allowedValues` https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2f48a42f8a0edf9a75a7f34e1e494eb355c5dbf2/src/FixerConfiguration/AllowedValueSubset.php#L34-L47

Since this just returns `false` and no context (e.g. exception with the actual missing value(s)), this isn't easy.

The actual exception is thrown from `\Symfony\Component\OptionsResolver\OptionsResolver` I learned -> https://github.com/symfony/options-resolver/blob/5d0f633f9bbfcf7ec642a2b5037268e61b0a62ce/OptionsResolver.php#L1076

The dilemma:
- the code in php-cs-fixer known the value but not the context (e.g. `'token'`)
- the code in symfony knows the context but not the value

If I would just throw from `\PhpCsFixer\FixerConfiguration\AllowedValueSubset::__invoke`, the output would become:
```
Loaded config default from "/Users/user/src/php-cs-fixer/.php_cs.dist".

In AbstractFixer.php line 170:

  [no_extra_blank_lines] Invalid configuration: Unsupported value yolo


In AllowedValueSubset.php line 44:

  Unsupported value yolo


fix [--path-mode PATH-MODE] [--allow-risky ALLOW-RISKY] [--config CONFIG] [--dry-run] [--rules RULES] [--using-cache USING-CACHE] [--cache-file CACHE-FILE] [--diff] [--diff-format DIFF-FORMAT] [--format FORMAT] [--stop-on-violation] [--show-progress SHOW-PROGRESS] [--] [<path>...]
```
but it's missing: `"Invalid configuration: The option "tokens" with value array is invalid`.

The only idea I've is that symfonys `OptionsResolver` should be able to catch exception from a callable and attach it to their own. Or so.

No idea, I don't think I've the expertise to provide a fix here. Especially if it requires changing Symfony also, I think I'll pass.

Thanks for the encouragement though :)
